File: src/main/java/io/koju/autopos/service/impl/ItemServiceImpl.java
package io.koju.autopos.service.impl;

import io.koju.autopos.service.ItemService;
import io.koju.autopos.domain.Item;
import io.koju.autopos.repository.ItemRepository;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import org.springframework.transaction.annotation.Transactional;
import org.springframework.stereotype.Service;

import javax.inject.Inject;
import java.util.List;
import java.util.Optional;

/**
 * Service Implementation for managing Item.
 */
@Service
@Transactional
public class ItemServiceImpl implements ItemService{

    private final Logger log = LoggerFactory.getLogger(ItemServiceImpl.class);
    
    @Inject
    private ItemRepository itemRepository;
    
    /**
     * Save a item.
     * @return the persisted entity
     */
    public Item save(Item item) {
        log.debug("Request to save Item : {}", item);
        Item result = itemRepository.save(item);
        return result;
    }

    /**
     *  get all the items.
     *  @return the list of entities
     */
    @Transactional(readOnly = true) 
    public Page<Item> findAll(Pageable pageable) {
        log.debug("Request to get all Items");
        Page<Item> result = itemRepository.findAll(pageable); 
        return result;
    }

    /**
     *  get one item by id.
     *  @return the entity
     */
    @Transactional(readOnly = true) 
    public Item findOne(Long id) {
        log.debug("Request to get Item : {}", id);
        Item item = itemRepository.findOneWithEagerRelationships(id);
        return item;
    }

    /**
     *  delete the  item by id.
     */
    public void delete(Long id) {
        log.debug("Request to delete Item : {}", id);
        itemRepository.delete(id);
    }
}

File: src/main/scala/com/kapilkoju/autopos/catalog/service/ItemServiceImpl.scala
package com.kapilkoju.autopos.catalog.service

import java.lang.Long

import com.kapilkoju.autopos.catalog.domain.Item
import org.slf4j.LoggerFactory
import org.springframework.data.domain.{Page, Pageable}
import org.springframework.stereotype.Service
import org.springframework.transaction.annotation.Transactional

@Service
@Transactional
class ItemServiceImpl(private val itemRepo: ItemRepo)
  extends ItemService {

  private val log = LoggerFactory.getLogger(classOf[ItemServiceImpl])

  override def save(item: Item): Item = {
    log.debug("Request to save Item : {}", item)
    itemRepo.save(item)
  }

  @Transactional(readOnly = true)
  override def findAll(query: String, pageable: Pageable): Page[Item] = {
    log.debug("Request to get all Items")

    if (query.isEmpty) {
      itemRepo.findAll(pageable)
    } else {
      itemRepo.findByCodeIgnoreCaseContainingOrNameIgnoreCaseContaining(query, query, pageable)
    }
  }

  @Transactional(readOnly = true)
  override def findOne(id: Long): Item = {
    log.debug("Request to get Item : {}", id)
    itemRepo.findOneWithEagerRelationships(id)
  }

  override def delete(id: Long): Unit = {
    log.debug("Request to delete Item : {}", id)
    itemRepo.delete(id)
  }

  override def adjustQuantity(item: Item, number: Integer): Unit = {
    val dbItem = itemRepo.findOne(item.getId)
    dbItem.setQuantity(dbItem.getQuantity + number)
    itemRepo.save(dbItem)
  }
}
